dao/elasticsearch/filesystem.go

```diff
@@ -335,3 +342,45 @@ func (dao *ControlPlaneDao) ReadyDFS(serviceID string, _ *int) (err error) {
 	err = dao.facade.ResetLock(ctx, serviceID)
 	return
 }
+
+// TagSnapshot tags an existing snapshot with 1 or more strings
+func (dao *ControlPlaneDao) TagSnapshot(request model.TagSnapshotRequest, newTagList *[]string) error {
+	ctx := datastore.Get()
+
+	// synchronize the dfs
+	dfslocker := dao.facade.DFSLock(ctx)
```

Comment:
Don't use DFSLock here...if you need to add a Tag to a snapshot use the volume level lock instead